flask/testing.py

```diff
@@ -20,7 +21,7 @@
     from urlparse import urlsplit as url_parse
 
 
-def make_test_environ_builder(app, path='/', base_url=None, *args, **kwargs):
+def make_test_environ_builder(app, path='/', base_url=None, json=None, *args, **kwargs):
```

Comment:
i agree, but the method signature becomes less explicit, and relies pretty heavily on implicit knowledge of the workings .

besides  look up at url_parse